dataprofiler/data_readers/data_utils.py

```diff
@@ -439,52 +441,124 @@ def read_csv_df(
     return data
 
 
-def read_parquet_df(
+def convert_unicode_col_to_utf8(input_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Convert all unicode columns in input dataframe to utf-8.
+
+    :param input_df: input dataframe
+    :type input_df: pd.DataFrame
+    :return: corrected dataframe
+    :rtype: pd.DataFrame
+    """
+    # Convert all the unicode columns to utf-8
+    types = input_df.apply(lambda x: pd.api.types.infer_dtype(x.values, skipna=True))
+
+    mixed_and_unicode_cols = types[types == "unicode"].index.union(
+        types[types == "mixed"].index
+    )
+
+    for col in mixed_and_unicode_cols:
+        input_df[col] = input_df[col].apply(
+            lambda x: x.encode("utf-8").strip() if isinstance(x, str) else x
+        )
+        input_df[col] = input_df[col].apply(
+            lambda x: x.decode("utf-8").strip() if isinstance(x, bytes) else x
+        )
+    return input_df
```

Comment:
I think this could be a bit more simple code here .... a bit DRY-er

Reply:
changed locally, but need to discuss before commit

Reply:
leaving this part unchanged after meeting